### PR TITLE
use proof instead of root in signUserOpHash

### PR DESCRIPTION
**Pull request by protodev-rage**
# Description

As per https://github.com/bcnmy/scw-contracts/blob/SCW-V2-Modular-SA/contracts/smart-account/modules/SessionKeyManagerModule.sol#L79-L86 merkleProof should be getHexProof of keccack256 of leaf

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

not test extensively, just matched specs with contract

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/modules/src/SessionKeyManagerModule.ts [hunk 1]
```diff
@@ -142,6 +142,15 @@ export class SessionKeyManagerModule extends BaseValidationModule {
       throw new Error('sessionID or sessionValidationModule should be provided.')
     }
 
+    const leafDataHex = ethers.utils.keccak256(
```
Comment (livingrockrises) [inline packages/modules/src/SessionKeyManagerModule.ts:145]: I think it should be like below,

```
const leafData = hexConcat([
        hexZeroPad(ethers.utils.hexlify(validUntil), 6),
        hexZeroPad(ethers.utils.hexlify(validAfter), 6),
        hexZeroPad(mockSessionValidationModule.address, 20),
        sessionKeyData,
      ]);
```

**Review by livingrockrises — COMMENTED** [review]
reviewed

File: packages/modules/src/SessionKeyManagerModule.ts (hunk 1 above)
Comment (livingrockrises) [reply packages/modules/src/SessionKeyManagerModule.ts:145]:
```
const leafDataHex = hexConcat([
      hexZeroPad(ethers.utils.hexlify(sessionSignerData.validUntil), 6),
      hexZeroPad(ethers.utils.hexlify(sessionSignerData.validAfter), 6),
      hexZeroPad(sessionSignerData.sessionValidationModule, 20),
      sessionSignerData.sessionKeyData
    ])

    // Generate the padded signature with (validUntil,validAfter,sessionVerificationModuleAddress,validationData,merkleProof,signature)
    let paddedSignature = defaultAbiCoder.encode(
      ['uint48', 'uint48', 'address', 'bytes', 'bytes32[]', 'bytes'],
      [
        sessionSignerData.validUntil,
        sessionSignerData.validAfter,
        sessionSignerData.sessionValidationModule,
        sessionSignerData.sessionKeyData,
        this.merkleTree.getHexProof(leafDataHex),
        signature
      ]
    )
```

**Comment by livingrockrises** [comment]
can you also change the base to main feat branch - feat/modular-SA-V2-integration

File: packages/modules/src/SessionKeyManagerModule.ts (hunk 1 above)
Comment (livingrockrises) [reply packages/modules/src/SessionKeyManagerModule.ts:145]: edited above code.

Comment (protodev-rage) [reply packages/modules/src/SessionKeyManagerModule.ts]: fixed in https://github.com/bcnmy/biconomy-client-sdk/pull/264/commits/b3da05f2e9c56973e96d0a7a3bc065aef23f9c18

**Review by livingrockrises — APPROVED** [review]
LGTM

**Pull request merged**